src/sentry/models/artifactbundle.py

```diff
@@ -133,32 +133,28 @@ class Meta:
         app_label = "sentry"
         db_table = "sentry_artifactbundleindex"
 
-        # TODO: this index can be removed and maybe replaced with a different one
-        # The `ReleaseFile` table has a `release_id+name` index. Similarly, we could
-        # create a `artifact_bundle+url` index, though the effectiveness of that
-        # might be limited as we are primarily doing substring searches.
-        index_together = (
-            ("organization_id", "release_name", "dist_name", "url", "artifact_bundle"),
-        )
+        index_together = (("url", "artifact_bundle"),)
 
 
 @region_silo_only_model
 class ReleaseArtifactBundle(Model):
     __include_in_export__ = False
 
     organization_id = BoundedBigIntegerField(db_index=True)
-    release_name = models.CharField(max_length=250, db_index=True)
+    release_name = models.CharField(max_length=250)
     # We use "" in place of NULL because the uniqueness constraint doesn't play well with nullable fields, since
     # NULL != NULL.
-    dist_name = models.CharField(max_length=64, default=NULL_STRING, db_index=True)
+    dist_name = models.CharField(max_length=64, default=NULL_STRING)
     artifact_bundle = FlexibleForeignKey("sentry.ArtifactBundle")
     date_added = models.DateTimeField(default=timezone.now)
 
     class Meta:
         app_label = "sentry"
         db_table = "sentry_releaseartifactbundle"
 
-        unique_together = (("organization_id", "release_name", "dist_name", "artifact_bundle"),)
+        # We add the organization_id to this index since there are many occurrences of the same release/dist
+        # pair, and we would like to reduce the result set by scoping to the org.
+        index_together = (("organization_id", "release_name", "dist_name", "artifact_bundle"),)
```

Comment:
What sorts of queries do you need to make here? Will you be querying on `organization_id, dist_name` without a `release_name` filter at all?

Reply:
The reasoning for this is that we have a lot of equal `release/dist` pairs across orgs, thus we would benefit from scoping down the org directly in the index.

Reply:
How we would query is (simplified):
```sql
SELECT *
FROM sentry_releaseartifactbundle AS t1 JOIN sentry_projectartifactbundle AS t2 JOIN sentry_artifactbundle AS t3
WHERE t1.organization_id = x AND t1. release_name = y AND t1.dist_name = z AND t2.project_id = a
```

In addition, the reasoning for including the artifact bundle in the composite index is to leverage a single index for both joining and filtering since in other queries we assume the dbms will choose to first filter out `ArtifactBundle` and then join it on the other tables with the `artifact_bundle_id` and the other fields specific to those tables (e.g., `release_name`, `url`, `debug_id`).

Reply:
In particular, I updated some of the queries explicitly to use that specific index:
https://github.com/getsentry/sentry/pull/52611

Initially, the queries performed a lot worse, but I figured out there was a skew that we were disproportionately querying orgs/releases with a ton of bundles.

Now that this is fixed, the optimized query performs better, clocking in at around ⅔ the *average* time as the query without using the composite index.

Reply:
Sounds good, just wanted to make sure you never needed to query on `release_name, artifact_bundle` etc, since that'd be a different index